File: scripts/gen_session.py
import sys, json, urllib.request

# Read keys from env
with open(r'C:\Users\claw\Desktop\Pipz\apps\web\.env.production', 'r') as f:
    content = f.read()

lines = content.strip().split('\n')
service_key = anon_key = ''
for line in lines:
    if line.startswith('SUPABASE_SERVICE_ROLE_KEY='):
        service_key = line.split('=', 1)[1]
    elif line.startswith('NEXT_PUBLIC_SUPABASE_ANON_KEY='):
        anon_key = line.split('=', 1)[1].strip('"')

url = "https://mxbuffmxvyuioidjzaet.supabase.co"

# 1. Generate a fresh OTP (magic link)
req1 = urllib.request.Request(
    f'{url}/auth/v1/admin/generate_link',
    data=json.dumps({"type": "magiclink", "email": "pipztest@gmail.com"}).encode(),
    headers={
        'apikey': anon_key,
        'Authorization': f'Bearer {service_key}',
        'Content-Type': 'application/json',
    },
    method='POST'
)
with urllib.request.urlopen(req1) as r:
    data = json.loads(r.read())
    otp = data['email_otp']
    print(f"OTP: {otp}")

# 2. Verify OTP to get session
req2 = urllib.request.Request(
    f'{url}/auth/v1/verify',
    data=json.dumps({"type": "magiclink", "token": otp, "email": "pipztest@gmail.com"}).encode(),
    headers={'apikey': anon_key, 'Content-Type': 'application/json'},
    method='POST'
)
with urllib.request.urlopen(req2) as r:
    session = json.loads(r.read())
    print(f"SESSION_OK: true")
    print(f"ACCESS_TOKEN: {session['access_token'][:20]}...")
    print(f"REFRESH_TOKEN: {session['refresh_token'][:20]}...")
    print(f"USER_ID: {session['user']['id']}")
    
    # Output tokens for use in browser
    print(f"\n--- BROWSER INJECT ---")
    token_data = {
        "access_token": session['access_token'],
        "refresh_token": session['refresh_token'],
        "user": session['user']
    }
    print(f"TOKEN_JSON: {json.dumps(token_data)}")

File: scripts/gen_session.js
const { createClient } = require('@supabase/supabase-js');
const fs = require('fs');

// Read the service role key
const envContent = fs.readFileSync('.env.production', 'utf8');
const match = envContent.match(/SUPABASE_SERVICE_ROLE_KEY="([^"]+)"/);
if (!match) { console.error("Key not found"); process.exit(1); }
const srk = match[1];
console.log('Key length:', srk.length);

const supabase = createClient(
  'https://mxbuffmxvyuioidjzaet.supabase.co',
  srk,
  { auth: { autoRefreshToken: false, persistSession: false } }
);

supabase.auth.admin.createSession({
  user_id: '0ca58feb-4714-4695-9395-d846f869126f'
}).then(r => {
  if (r.data?.session) {
    console.log('SESSION SUCCESS');
    console.log('access_token:', r.data.session.access_token?.slice(0, 50) + '...');
    console.log('refresh_token:', r.data.session.refresh_token?.slice(0, 50) + '...');
    console.log('Full session:', JSON.stringify(r.data.session, null, 2).slice(0, 1000));
  } else {
    console.log('ERROR:', JSON.stringify(r.error));
  }
}).catch(e => console.log('EXCEPTION:', e.message));
